internal/pkg/epinject/ociregistry/server.go

```diff
@@ -600,8 +600,19 @@ func (s *Server) updateManifest(w http.ResponseWriter, r *http.Request) {
 		Author:    "waypoint",
 	})
 
-	config.Config.Entrypoint = append([]string{"/waypoint-entrypoint"}, config.Config.Entrypoint...)
-	s.Logger.Debug("injected entrypoint", "value", config.Config.Entrypoint)
+	var found bool
+	for _, entryPoint := range config.Config.Entrypoint {
+		if entryPoint == "/waypoint-entrypoint" {
+			found = true
+			s.Logger.Debug("waypoint-entrypoint found, not injecting")
+			break
```

Comment:
Intentionally didn't use the new `containsEntrypoint` function here to avoid including a new package (which at one point gave me a circular dependency issue, but that may be resolved)